lib/pages/chat/send_file_dialog/send_file_dialog.dart

```diff
@@ -0,0 +1,209 @@
+import 'package:fluffychat/pages/chat/input_bar/focus_suggestion_controller.dart';
+import 'package:fluffychat/pages/chat/send_file_dialog/send_file_dialog_view.dart';
+import 'package:fluffychat/presentation/extensions/send_file_web_extension.dart';
+import 'package:fluffychat/presentation/list_notifier.dart';
+import 'package:fluffychat/utils/matrix_sdk_extensions/matrix_file_extension.dart';
+import 'package:fluffychat/presentation/enum/chat/send_media_with_caption_status_enum.dart';
+import 'package:flutter/material.dart';
+
+import 'package:matrix/matrix.dart';
+
+class SendFileDialog extends StatefulWidget {
+  final Room? room;
+  final List<MatrixFile> files;
+
+  const SendFileDialog({
+    this.room,
+    required this.files,
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  SendFileDialogController createState() => SendFileDialogController();
+}
+
+class SendFileDialogController extends State<SendFileDialog> {
+  final sendMediaOnWebWithCaptionInteractor =
+      getIt.get<SendMediaOnWebWithCaptionInteractor>();
+
+  final sendFilesOnWebWithCaptionInteractor =
+      getIt.get<SendFilesOnWebWithCaptionInteractor>();
+
+  final focusSuggestionController = FocusSuggestionController();
+
+  final TextEditingController textEditingController = TextEditingController();
+
+  final FocusNode captionsFocusNode = FocusNode();
+
+  final ValueKey sendFileDialogTypeAheadKey =
+      const ValueKey('sendFileDialogTypeAhead');
+
+  bool isSendMediaWithCaption = true;
+
+  ListNotifier<MatrixFile> filesNotifier = ListNotifier([]);
+
+  Map<MatrixFile, MatrixImageFile?> thumbnails = {};
+
+  ValueNotifier<double> maxMediaSizeNotifier = ValueNotifier(double.infinity);
+
+  ValueNotifier<bool> haveErrorFilesNotifier = ValueNotifier(false);
+
+  @override
+  void initState() {
+    super.initState();
+    filesNotifier = ListNotifier(widget.files);
+    isSendMediaWithCaption = _isShowSendMediaDialog(
+      filesNotifier.value,
+      widget.room,
+    );
+    requestFocusCaptions();
+    WidgetsBinding.instance.addPostFrameCallback((timeStamp) async {
+      await loadMaxMediaSize();
+      await loadThumbnailsForMedia();
+    });
+  }
+
+  @override
+  void dispose() {
+    textEditingController.dispose();
+    focusSuggestionController.dispose();
+    captionsFocusNode.dispose();
+    filesNotifier.dispose();
+    maxMediaSizeNotifier.dispose();
+    haveErrorFilesNotifier.dispose();
+    super.dispose();
+  }
+
+  Future<void> loadMaxMediaSize() async {
+    if (widget.room == null) {
+      return;
+    }
+    final mediaConfig = await widget.room?.client.getConfig();
+    maxMediaSizeNotifier.value =
+        mediaConfig?.mUploadSize?.toDouble() ?? double.infinity;
+    updateHaveErrorFilesNotifier();
+  }
+
+  void updateHaveErrorFilesNotifier() {
+    haveErrorFilesNotifier.value = filesNotifier.value.any(
+      (file) => file.size > maxMediaSizeNotifier.value,
+    );
+  }
+
+  Future<void> loadThumbnailsForMedia() async {
```

Comment:
Why not load thumbnail in item builder? IMO, we can do the thumbnail for each file? not do for all and update the view as `PUM, SURPRISE`

Reply:
because that would need a `FutureBuilder`, and everytime we remove an item from list, the list got updated then rebuild. It makes the whole list blink